src/dom-to-image.js

```diff
@@ -244,6 +259,86 @@
                         });
                     }
                 }
+
+                if(domtoimage.impl.options.scrollFix &&
+                    (original.scrollTop || original.scrollLeft)) {
+                    // Setup container for absolute positioning of children
+                    clone.style.position = 'relative';
+                    clone.style.overflow = 'hidden';
+                    clone.style.width = original.offsetWidth + 'px';
+                    clone.style.height = original.offsetHeight + 'px';
+                    var scrollTopRemaining = original.scrollTop > 0 ? original.scrollTop : null;
+                    var scrollLeftRemaining = original.scrollLeft > 0 ? original.scrollLeft : null;
+                    var originalIsPositionRelative = originalStyle['position'] === 'relative';
+                    var computedStylesCache = {};
```

Comment:
Why are these objects `{}` instead of arrays `[]` given we index them by `i` everywhere?

Reply:
Could be arrays since `i` should always be an integer, I didnt have a particular reason here for not using an array other than I usually use a map/dict for this kind of caching. Is there any benefit of an array vs an object/dict/map here?